File: app/src/main/java/mb00/android/codehub/ui/search/inject/SearchUsersFragmentModule.java
package mb00.android.codehub.ui.search.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.search.interactor.SearchInteractor;
import mb00.android.codehub.ui.search.router.SearchRouter;
import mb00.android.codehub.ui.search.viewmodel.SearchUsersViewModel;


@Module
public class SearchUsersFragmentModule {

    @Provides
    SearchUsersViewModel provideSearchUsersViewModel(SearchInteractor interactor, SearchRouter router) {
        return new SearchUsersViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/search/inject/SearchUsersFragmentModule.kt
package mb00.android.codehub.ui.search.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.search.interactor.SearchInteractor
import mb00.android.codehub.ui.search.router.SearchRouter
import mb00.android.codehub.ui.search.viewmodel.SearchUsersViewModel


@Module
class SearchUsersFragmentModule {

    @Provides
    internal fun provideSearchUsersViewModel(interactor: SearchInteractor, router: SearchRouter): SearchUsersViewModel {
        return SearchUsersViewModel(interactor, router)
    }

}
